packages/solutions/app-tools/src/builder/shared/bundlerPlugins/RouterPlugin.ts

```diff
@@ -276,30 +207,32 @@ export class RouterPlugin {
             })();
           `;
 
-            const source = chunkToSource.get(chunkId);
-            const map = chunkToMap.get(chunkId);
-            const newContent = `${injectedContent}${source?.toString()}`;
-
-            const result = await transform(newContent, {
-              loader: path.extname(file).slice(1) as Loader,
-              sourcemap: true,
-              minify: minimize,
-            });
-
-            const newSource = new SourceMapSource(
-              result.code,
-              file,
-              result.map,
-              source?.toString(),
-              map,
-            );
+            let htmlName;
+            for (const [chunks, name] of chunksToHtmlName.entries()) {
+              if (
+                Array.isArray(chunkNames) &&
+                Array.isArray(chunks) &&
+                chunkNames.every((value, index) => value === chunks[index])
+              ) {
+                htmlName = name;
+                break;
+              }
+            }
 
-            compilation.updateAsset(
-              file,
-              newSource,
-              // FIXME: The arguments third of updatgeAsset is a optional function in webpack.
-              undefined as any,
-            );
+            if (htmlName) {
+              const oldHtml = compilation.assets[htmlName];
+              compilation.updateAsset(
+                htmlName,
+                new RawSource(
+                  oldHtml
+                    .source()
+                    .toString()
+                    .replace(placeHolder, injectedContent),
```

Comment:
Can we provide an option to disable the inline script? Some businesses may not want inline scripts.

Reply:
This may have some issues. this plugin will only inject manifests,  what scenarios do they not want inline scripts?